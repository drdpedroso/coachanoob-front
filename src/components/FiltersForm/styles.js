import styled from 'styled-components'
import {Form} from "semantic-ui-react"

export const FiltersContainer = styled.div`
  height: 40px;
  width: 100%; 
  box-shadow: 0 1px 5px 0 rgba(0,0,0,.2), 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.12);
  margin-bottom: 32px;
`

export const Label = styled.label`
  color: white !important;
`

export const FormContainer = styled(Form)`
    max-width: 600px;
     margin: auto;
`
export const FiltersInnerContainer = styled.div`
  background-color: #5063F0;
  padding: 20px;
  display: flex; 
  flex-direction: column
`
