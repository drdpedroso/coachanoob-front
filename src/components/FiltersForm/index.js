import React, {useState} from 'react'
import PropTypes from 'prop-types';
import Button from '../Button'
import {Form, Responsive, Segment} from 'semantic-ui-react'
import BaseDropdown from "../Dropdown"
import BaseInput from "../Input"
import RadioButtonGroup from "../RadioButtonGroup"
import {FiltersContainer, FormContainer, Label, FiltersInnerContainer} from './styles'
import StickyButton from "../StickyButton";

const FiltersForm = (props) => {
    const [form, setForm] = useState({})
    const [open, setOpen] = useState(false)

    const onChangeDropdownValue = (value: string, key: string) => {
        setForm({
            ...form,
            [key]: value
        })
    }

    const onChangeInputValue = (e: any, key: string) => {
        const {value} = e.target
        setForm({
            ...form,
            [key]: value
        })
    }
    const onSubmit = (): function => {
        return typeof props.onSubmit === 'function' ? props.onSubmit(form) : null
    }

    const openFilters = () => setOpen(!open)

    const items = [
        {name: '1+', value: '1'},
        {name: '2+', value: '2'},
        {name: '3+', value: '3'},
        {name: '4+', value: '4'},
        {name: '5+', value: '5'},
    ]
    const countryOptions = [
        {key: 'af', value: 'af', flag: 'af', text: 'Afghanistan'},
        {key: 'ax', value: 'ax', flag: 'ax', text: 'Aland Islands'},
        {key: 'al', value: 'al', flag: 'al', text: 'Albania'},
        {key: 'dz', value: 'dz', flag: 'dz', text: 'Algeria'},
        {key: 'as', value: 'as', flag: 'as', text: 'American Samoa'},
        {key: 'ad', value: 'ad', flag: 'ad', text: 'Andorra'},
        {key: 'ao', value: 'ao', flag: 'ao', text: 'Angola'},
        {key: 'ai', value: 'ai', flag: 'ai', text: 'Anguilla'},
        {key: 'ag', value: 'ag', flag: 'ag', text: 'Antigua'},
        {key: 'ar', value: 'ar', flag: 'ar', text: 'Argentina'},
        {key: 'am', value: 'am', flag: 'am', text: 'Armenia'},
        {key: 'aw', value: 'aw', flag: 'aw', text: 'Aruba'},
        {key: 'au', value: 'au', flag: 'au', text: 'Australia'},
        {key: 'at', value: 'at', flag: 'at', text: 'Austria'},
        {key: 'az', value: 'az', flag: 'az', text: 'Azerbaijan'},
        {key: 'bs', value: 'bs', flag: 'bs', text: 'Bahamas'},
        {key: 'bh', value: 'bh', flag: 'bh', text: 'Bahrain'},
        {key: 'bd', value: 'bd', flag: 'bd', text: 'Bangladesh'},
        {key: 'bb', value: 'bb', flag: 'bb', text: 'Barbados'},
        {key: 'by', value: 'by', flag: 'by', text: 'Belarus'},
        {key: 'be', value: 'be', flag: 'be', text: 'Belgium'},
        {key: 'bz', value: 'bz', flag: 'bz', text: 'Belize'},
        {key: 'bj', value: 'bj', flag: 'bj', text: 'Benin'},
    ]
    return (
        <div>

            <Responsive as={'div'} maxWidth={768}>
                <StickyButton onClick={openFilters} data-testid="open-filters">Abrir</StickyButton>
            </Responsive>
            <Responsive as={'div'} minWidth={769}>
                <FiltersContainer>
                    <Button onClick={openFilters} data-testid="open-filters">Abrir</Button>
                </FiltersContainer>
            </Responsive>

            {open && <FiltersInnerContainer>
                <FormContainer>
                    <Form.Field>
                        <Label>Tipo de Uso</Label>
                        <BaseDropdown options={countryOptions}
                                      onChange={(e, data) => onChangeDropdownValue(data.value, 'usageType')}
                                      selection/>
                    </Form.Field>
                    <Form.Field>
                        <Label>Tipo de Im√≥vel</Label>
                        <BaseDropdown options={countryOptions}
                                      onChange={(e, data) => onChangeDropdownValue(data.value, 'type')} selection/>
                    </Form.Field>
                    <Form.Group>
                        <Form.Field width={4}>
                            <Label>Estado</Label>
                            <BaseDropdown fluid selection defaultValue="SP"
                                          onChange={(e, data) => onChangeDropdownValue(data.value, 'state')}/>
                        </Form.Field>
                        <Form.Field width={12}>
                            <Label>Cidade</Label>
                            <BaseDropdown fluid selection
                                          onChange={(e, data) => onChangeDropdownValue(data.value, 'city')}/>
                        </Form.Field>
                    </Form.Group>
                    <Form.Field>
                        <Label>Bairro</Label>
                        <BaseInput data-testid="neighborhood" disabled
                                   onChange={(e) => onChangeInputValue(e, 'neighborhood')}/>
                    </Form.Field>
                    <Form.Group widths="equal">
                        <Form.Field>
                            <Label>Valor min.</Label>
                            <BaseInput data-testid="salePriceFrom" fluid
                                       onChange={(e) => onChangeInputValue(e, 'salePriceFrom')}/>
                        </Form.Field>
                        <Form.Field>
                            <Label>Valor max.</Label>
                            <BaseInput data-testid="salePriceTo" fluid
                                       onChange={(e) => onChangeInputValue(e, 'salePriceTo')}/>
                        </Form.Field>
                    </Form.Group>
                    <Form.Field>
                        <Label>Quartos</Label>
                        <RadioButtonGroup data-testid="rooms" white
                                          onItemClick={(value) => onChangeDropdownValue(value, 'rooms')} items={items}/>
                    </Form.Field>
                    <Form.Field>
                        <Label>Suites</Label>
                        <RadioButtonGroup data-testid="suites" white
                                          onItemClick={(value) => onChangeDropdownValue(value, 'suites')}
                                          items={items}/>
                    </Form.Field>
                    <Form.Field>
                        <Label>Banheiros</Label>
                        <RadioButtonGroup white onItemClick={(value) => onChangeDropdownValue(value, 'rooms')}
                                          items={items}/>
                    </Form.Field>
                    <Form.Field>
                        <Label>Vagas na Garagem</Label>
                        <RadioButtonGroup white onItemClick={(value) => onChangeDropdownValue(value, 'rooms')}
                                          items={items}/>
                    </Form.Field>
                    <Button data-testid="submit" type='submit' fluid onClick={onSubmit}>Filtrar</Button>
                </FormContainer>
            </FiltersInnerContainer>}
        </div>
    )
}
FiltersForm.propTypes = {
    onClick: PropTypes.func,
    children: PropTypes.node
};

export default FiltersForm
