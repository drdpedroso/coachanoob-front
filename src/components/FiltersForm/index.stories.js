import React from 'react';
import {storiesOf} from '@storybook/react';
import {action} from '@storybook/addon-actions';

import FiltersForm from './';
import {Icon, Segment} from "semantic-ui-react";

storiesOf('FiltersForm', module)
    .add('with text primary', () => (
        <FiltersForm/>
    ))

