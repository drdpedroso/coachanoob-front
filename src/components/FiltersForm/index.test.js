import React from 'react';
import FiltersForm from './index'
import renderer from 'react-test-renderer';
import {cleanup} from "@testing-library/react"

afterEach(cleanup)

describe('FiltersForm', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<FiltersForm/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
