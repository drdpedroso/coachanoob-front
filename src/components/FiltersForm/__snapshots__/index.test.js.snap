// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersForm renders correctly 1`] = `
<div>
  <div
    className="sc-htpNat icOVtr"
  >
    <button
      className="ui button"
      data-testid="open-filters"
      onClick={[Function]}
    >
      Abrir
    </button>
  </div>
</div>
`;
